spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSnapshotUpdateSparkAction.java

```diff
@@ -38,10 +39,12 @@ public ThisT snapshotProperty(String property, String value) {
 
   protected void commit(org.apache.iceberg.SnapshotUpdate<?> update) {
     summary.forEach(update::set);
+    update.set(SparkWriteUtil.SPARK_APP_ID, spark().sparkContext().applicationId());
```

Comment:
all of this is already automatically handled via `EnvironmentContext`. It's handled at https://github.com/apache/iceberg/blob/3e0f7a1bfb167c4d18ebdc9d0f69afe0f4754512/spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L778 and then added to the summary in https://github.com/apache/iceberg/blob/be577eeac631d77243beb57409e476bf197f79d7/core/src/main/java/org/apache/iceberg/SnapshotProducer.java#L409